### PR TITLE
add basic ci

### PR DESCRIPTION
**Pull request by cmeissl**
based on #6

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  format:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
```
Comment (YaLTeR) [inline .github/workflows/ci.yml:16]: Checkout is at v4 nowadays

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      - name: Rust toolchain
+        uses: actions-rs/toolchain@v1
```
Comment (YaLTeR) [inline .github/workflows/ci.yml:18]: actions-rs actions are long unmaintained; https://github.com/dtolnay/rust-toolchain is the replacement

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          default: true
+      - name: Cargo cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-rust_stable-${{ hashFiles('**/Cargo.toml') }}
+      - name: Format
+        uses: actions-rs/cargo@v1
```
Comment (YaLTeR) [inline .github/workflows/ci.yml:33]: And for this just a `cargo fmt --all --check` shell line

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cmeissl) [reply .github/workflows/ci.yml]: thx, updated to use v4

File: .github/workflows/ci.yml (hunk 2 above)
Comment (cmeissl) [reply .github/workflows/ci.yml]: right, updated to use rust-toolchain

**Pull request merged**